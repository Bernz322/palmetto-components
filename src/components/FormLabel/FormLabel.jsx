import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './FormLabel.scss';

const FormLabel = ({
  error,
  inputId,
  isFieldRequired,
  labelText,
}) => {
  const labelClasses = classNames(
    error ? 'error' : null,
    'label',
  );

  return (
    <label className={labelClasses} htmlFor={inputId}>
      {labelText}
      {isFieldRequired && <span className="font-size-sm">&nbsp;*</span>}
    </label>
  );
};

FormLabel.propTypes = {
  error: PropTypes.bool,
  inputId: PropTypes.string.isRequired,
  labelText: PropTypes.string.isRequired,
  isFieldRequired: PropTypes.bool,
};

FormLabel.defaultProps = {
  error: false,
  isFieldRequired: false,
};

export default FormLabel;
