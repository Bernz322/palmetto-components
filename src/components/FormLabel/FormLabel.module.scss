.label {
  display: block;
  margin: 0;
  padding: 0;
  color: var(--color-font-base);
  font-size: var(--size-font-sm);
  font-weight: 700;

  &.inline {
    display: inline-block;
  }

  &.radio-input-label {
    font-weight: 400;
  }

  &.disabled {
    color: var(--color-font-grey-light);
  }

  &.error {
    color: var(--color-brand-danger-base);

    &.disabled {
      color: var(--color-brand-danger-lighter);
    }
  }

  +input,
  +.select-input {
    margin-top: var(--size-spacing-xs);
  }
}
