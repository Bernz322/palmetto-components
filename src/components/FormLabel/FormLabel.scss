@import '../../styles/variables.scss';

.label {
  font-size: $size-font-sm;
  color: $color-font-base;
  display: block;
  font-weight: 700;

  &.error {
    color: $color-brand-danger-base;
  }
}

.label+input[type=text],
.label+input[type=tel],
.label+input[type=password],
.label+input[type=search],
.label+input[type=url] {
  margin-top: $size-spacing-xs;
}
