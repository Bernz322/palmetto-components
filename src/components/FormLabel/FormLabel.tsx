import React from 'react';
import PropTypes, { InferProps } from 'prop-types';
import classNames from 'classnames';
import './FormLabel.scss';

const propTypes = {
  /**
   * Mark the label has invalid
   */
  hasError: PropTypes.bool.isRequired,
  /**
   * The id of the form control that the label is labeling
   */
  inputId: PropTypes.string.isRequired,
  /**
   * The label text
   */
  labelText: PropTypes.node.isRequired,
  /**
   * Render an asterisk after the label to mark it as required
   */
  isFieldRequired: PropTypes.bool.isRequired,
};


const FormLabel = ({
  hasError,
  inputId,
  isFieldRequired,
  labelText,
}: InferProps<typeof propTypes>) => {
  const labelClasses = classNames(
    'label',
    {
      error: hasError,
    },
  );

  return (
    <label className={labelClasses} htmlFor={inputId}>
      {labelText}
      {isFieldRequired && <span className="font-size-sm">&nbsp;*</span>}
    </label>
  );
};

FormLabel.propTypes = propTypes;

FormLabel.defaultProps = {
  hasError: false,
  isFieldRequired: false,
};

export default FormLabel;
