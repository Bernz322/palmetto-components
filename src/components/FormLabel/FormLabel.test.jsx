import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import FormLabel from './FormLabel';

beforeEach(() => {
  console.error = jest.fn(); // eslint-disable-line no-console
});

afterEach(() => {
  console.error.mockRestore(); // eslint-disable-line no-console
});

describe('FormLabel', () => {
  test('Throws error if required prop "inputId" is not supplied to component', () => {
    render(<FormLabel labelText="hello" />);
    expect(console.error).toHaveBeenCalledTimes(1); // eslint-disable-line no-console
    expect(console.error.mock.calls[0][0]) // eslint-disable-line no-console
      .toContain('Failed prop type: The prop `inputId`');
  });

  test('Throws error if required prop "labelText" is not supplied to component', () => {
    render(<FormLabel inputId="myId" />);
    expect(console.error).toHaveBeenCalledTimes(1); // eslint-disable-line no-console
    expect(console.error.mock.calls[0][0]) // eslint-disable-line no-console
      .toContain('Failed prop type: The prop `labelText`');
  });

  test('Matches the snapshot for component with default props', () => {
    const { asFragment } = render(<FormLabel inputId="myId" labelText="hello world" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Matches the snapshot for component with isRequired prop', () => {
    const { asFragment } = render(<FormLabel inputId="myId" labelText="hello world" isRequired />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Matches the snapshot for component with hasError prop', () => {
    const { asFragment } = render(<FormLabel inputId="myId" labelText="hello world" hasError />);
    expect(asFragment()).toMatchSnapshot();
  });
});
