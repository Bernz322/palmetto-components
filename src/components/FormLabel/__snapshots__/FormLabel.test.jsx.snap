// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLabel Matches the snapshot for component with default props 1`] = `
<DocumentFragment>
  <label
    class="Palmetto-FormLabel"
    for="myId"
  >
    hello world
  </label>
</DocumentFragment>
`;

exports[`FormLabel Matches the snapshot for component with hasError prop 1`] = `
<DocumentFragment>
  <label
    class="Palmetto-FormLabel error"
    for="myId"
  >
    hello world
  </label>
</DocumentFragment>
`;

exports[`FormLabel Matches the snapshot for component with isRequired prop 1`] = `
<DocumentFragment>
  <label
    class="Palmetto-FormLabel"
    for="myId"
  >
    hello world
  </label>
</DocumentFragment>
`;
